### PR TITLE
Missing props in DeckProps, optional width and height

### PR DESCRIPTION
**Pull request by Lock-up**
Added properties:
 ```
parent?: HTMLElement;
pickable?: boolean;
```
Changed width and height to be optional:
```
width?: string | number;
height?: string | number;
```

### CODE REVIEW

**Review by strican — COMMENTED** [review]
Sorry for the novice review, but I can't find anything that guarantees what props are available at any given time in `Deck.props`, nor that are required in `Deck.constructor`. Would it be better to just make everything here non-optional and then edit the `Deck` class to use `Partial<DeckProps`, as it already does in the `setProps` method?

I'm very open to being wrong about this - as I said, I'm somewhat new to DeckGL - but I just couldn't find these guarantees in the docs.

File: deck.gl__core/index.d.ts [hunk 1]
```diff
@@ -2354,8 +2354,8 @@ declare module "@deck.gl/core/lib/deck" {
 	export interface DeckProps {
 		//https://deck.gl/#/documentation/deckgl-api-reference/deck?section=properties
```
[the remark this answers is not in the document]
Comment (danmarshall) [reply deck.gl__core/index.d.ts:2355]: Yes, that would be good, old one is now broken :|

[the remark this answers is not in the document]
Comment (Lock-up) [reply deck.gl__core/index.d.ts]: Hi, thank you for your feedback.
I updated the link and made everything non optional and then used Partial<DeckProps> in the constructor

File: deck.gl__core/index.d.ts [hunk 2]
```diff
@@ -2409,23 +2410,23 @@ declare module "@deck.gl/core/lib/deck" {
...
+		onError: (error: Error, source: any) => void;
+		_onMetrics: (metrics: MetricsPayload) => void;
 
-		ContextProvider?: React.Provider<ContextProviderValue>
+		ContextProvider: React.Provider<ContextProviderValue>
 	}
 
 	export default class Deck {
-		constructor(props: DeckProps);
+		constructor(props: Partial<DeckProps>);
```
Comment (strican) [inline deck.gl__core/index.d.ts:2429]: Do we need to make `props` a `Partial` as well?

Comment (Lock-up) [reply deck.gl__core/index.d.ts:2429]: I think so, otherwise you have to provide values for all properties.

Comment (strican) [reply deck.gl__core/index.d.ts:2429]: Sorry, I meant the field `props` a few lines down in the `Deck` class, not the parameter here.

Comment (Lock-up) [reply deck.gl__core/index.d.ts:2429]: As far as I can see are all parameter either default or overridden in the constructor. So, they should all be set and `Partial<>` isn't needed.

Comment (strican) [reply deck.gl__core/index.d.ts:2429]: Thanks for looking into that!

**Review by strican — APPROVED** [review]
Looks good to my eye, my feedback was addressed

**Pull request merged**